shared_python/ValidateTags.py

```diff
@@ -0,0 +1,187 @@
+        else:
+            self.log.warning('Manual Input Failed. This field cannot be empty.')
+        print('\x1b[0m', end='')
+
+    def prompt_correction(self, tag_name, tag_type, isType):
+        print ('\rThe following values are accepted:')
+        if (isType):
+            print('\r' + self.list_dicts(None, tag_type))
+        else:
+            print('\r' + self.list_dicts(tag_name, tag_type))
+        prompt = '\rPlease enter the correct name for "' + tag_name + '"'
+        if (isType):
+            prompt += ' (Press ENTER to default to "tags"): '
+        else:
+            prompt += ': '
+        tag_correction = input(prompt)
+        if (isType and not tag_correction):
+            tag_correction = "tags"
+        return tag_correction
+
+    ## Self Correction Logic
+    ## NOTE: This is for very light corrections only.
+    def correct_tag_type(self, tag_type):
+        # Attempt self correciton by making whole word lowercase
+        tag_correction = tag_type.lower()
+        if (self.classify_tag(None, tag_correction) > 0):
+          self.print_tag_correction(tag_type, tag_correction)
+          return tag_correction
+
+        # Attempt self correction by adding last character
+        if (self.classify_tag(None, tag_correction + "s") > 0):
+          self.print_tag_correction(tag_type, tag_correction + "s")
+          return tag_correction + "s"
+
+        # Attempt self correction by removing last character
+        if (self.classify_tag(None, tag_correction[:-1]) > 0):
+          self.print_tag_correction(tag_type, tag_correction[:-1])
+          return tag_correction[:-1]
+
+        #All attempts failed
+        return None
+
+    def correct_tag(self, tag, tag_type):
+        # Attempt self correction by uppercasing every first character
+        if (self.classify_tag(tag.title(), tag_type) > 0):
+            self.print_tag_correction(tag, tag.title())
+            return tag.title()
+
+        # All attempts failed
+        return None
+
+    ## Gating/Switch Logic
+    def classify_tag(self, tag, tag_type):
+        if (not tag):
+            return self.identify_tag_type(tag_type)
+        if (tag_type == "rating"):
+            return self.identify_rating(tag)
+        if (tag_type == "categories"):
+            return self.identify_category(tag)
+        if (tag_type == "warnings"):
+            return self.identify_warning(tag)
+        return 20
+
+    def list_dicts(self, tag, tag_type):
+        if (not tag):
+            return str(list(self.dict_tag_type.keys()))
+        elif (tag_type == 'rating'):
+            return str(list(self.dict_rating.keys()))
+        elif (tag_type == 'categories'):
+            return str(list(self.dict_category.keys()))
+        elif (tag_type == 'warnings'):
+            return str(list(self.dict_warning.keys()))
+
+    ## Verify methods to use
+    def verify_tag_type(self, tag_type):
```

Comment:
More naming quibbling: does this method only verify the tag, or is it correcting it? Since this class is a "validator" and this method returns a corrected tag type, maybe `validate_and_fix_tag_type`? Or just `fix_tag_type` for brevity? (or `correct...` if you prefer that terminology which is what you're used throughout) - and same for `verify_tag` below.